### ISSUE TITLE
[Translator] Translator warmup doesn't dump fallback locales

### ISSUE THREAD

**Issue by kminh** [symfony/symfony#15364]
Using Symfony 2.7.2.

When fallback locales for the Translator service are set to `[en, fr]` for example, if I run `app/console cache:clear` (or via API), cached catalogues for `en` and `fr` are not dumped.

The line causing this issue is: https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/Translation/Translator.php#L338

It seems that fallback catalogues are generated on the fly to be added to other locales, thus being by passed when dumping.

**Comment by danielmitd** [comment]
Any easy workaround would be to move the default locale to at the beginning of the [resourceLocales](https://github.com/symfony/symfony/blob/2.7/src/Symfony/Bundle/FrameworkBundle/Translation/Translator.php#L85-L87) list. 

```php
$defaultLocale = $this->container->getParameter('kernel.default_locale');
$locales       = $this->resourceLocales;

// handle defaultLocale first, to prevent missing cache file generation
if (false !== ($key = array_search($defaultLocale, $locales))) {
    unset($locales[$key]);
    array_unshift($locales, $defaultLocale);
}

foreach ($locales as $locale) {
    $this->loadCatalogue($locale);
}
```

In this case the default locale catalogue is dumped first, and then reused during the dump of the over locales. But it is still a "workaround" :wink:

**Comment by aitboudad** [comment]
seems weird, I'll see what's going wrong :)

**Comment by aitboudad** [comment]
fixed by #15516
